cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/store/DefaultStoreTest.java

```diff
@@ -517,6 +518,88 @@ public void testUpdateChangedApplication() throws ConflictException {
     Assert.assertEquals(FooMapReduceJob.class.getName(), spec.getMapReduce().get("mrJob3").getClassName());
   }
 
+  @Test
+  public void testAddApplicationWithoutMarkingLatest()
+      throws ConflictException {
+    ApplicationId appId = new ApplicationId("account1", "app1");
+    ApplicationMeta appMeta = new ApplicationMeta("app1", Specifications.from(new FooApp()),
+        new ChangeDetail(null, null, null,
+            System.currentTimeMillis()), null, false);
```

Comment:
don't use values like current time in unit tests, as you can never compare them to expected values.

Reply:
Updated the tests to store the creation time as a variable (instead of passing the value directly) so that we can compare it with the actual value when needed.